### DISCUSSION TITLE
How to limit size of a specified path?

### DISCUSSION THREAD

**Discussion by UchihaYuki** [minio/minio#20742] [Q&A]
I allow users to upload files through the following policy:
```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Action": ["s3:GetObject", "s3:PutObject"],
      "Resource": ["arn:aws:s3:::user/${jwt:sub}/*"]
    }
  ]
}
```
I want to limit the total size of files a user can upload. I only find this command `mc admin bucket quota` to limit the size of the whole bucket.

**Comment by UchihaYuki** [comment]
I just found the following in [doc](https://min.io/docs/minio/linux/reference/deprecated/mc-quota-set.html):
> mc quota set
> Changed in version RELEASE.2022-12-13T00-23-28Z: mc quota set replaced mc admin bucket quota --hard.
>
>Changed in version RELEASE.2024-07-31T15-58-33Z: mc quota set is deprecated.

So, is there still a way to set size limitations?

**Reply by klauspost** [reply]
No.

**Reply by marktheunissen** [reply]
@UchihaYuki If you want this, you must presign the URLS, use https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-HTTPPOSTConstructPolicy.html

`content-length-range` is the parameter in the policy

**Reply by UchihaYuki** [reply]
> @UchihaYuki If you want this, you must presign the URLS, use https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-HTTPPOSTConstructPolicy.html
> 
> `content-length-range` is the parameter in the policy

Is there a simple way to limit the size of the objects uploaded to a specified bucket? Because I allow authenticated users to upload files, and they use AssumeRoleWithWebIdentity to login. I think it's better for me to give them some restrictions.

**Reply by klauspost** [reply]
@UchihaYuki No. It is not possible.

**Reply by UchihaYuki** [reply]
> @UchihaYuki No. It is not possible.

Then can I limit the size of objects in a MinIO deployment as a whole, so that I can deploy another MinIO instance for my users?

I see [the max object size is 50 TiB](https://min.io/docs/minio/linux/operations/concepts/thresholds.html), if there is no way to limit the users, it's dangerous.

**Reply by marktheunissen** [reply]
Not possible unfortunately.
